autoscheduler/frontend/src/redux/reducers.ts

```diff
@@ -61,10 +102,117 @@ function courseCards(
   }
 }
 
+// sets whether the user is selecting busy time or not
+// in the future, this could also support preferred times
+function availabilityMode(
+  state: AvailabilityType = AvailabilityType.NONE, action: AvailabilityModeAction,
+): AvailabilityType {
+  if (action.type === SET_AVAILABILITY_MODE) { return action.mode; }
+  return state;
+}
+
+// stores all information about the user's availability
+function availability(
+  state: Availability[] = [], action: AvailabilityAction,
+): Availability[] {
+  switch (action.type) {
+    case ADD_AVAILABILITY:
+    case MERGE_AVAILABILITY: {
+      // check for overlaps between the availability to be added and pre-existing ones
+      let overlapsFound = 0; // counts merging overlaps only
+      let overlapIdx = -1;
+
+      const initNewAv = action.type === ADD_AVAILABILITY
+        ? argsToAvailability(action.availability)
+        : state[state.length - 1];
+      let newAv = initNewAv;
```

Comment:
Shouldn't we set `newAv` to what `initNewAv` is? Since this is the only time we use `initNewAv`

Reply:
Fixed